src/SonarScanner.MSBuild.Tasks/GetAnalyzerSettings.cs

```diff
@@ -367,6 +388,30 @@ private static bool IsAssemblyLibraryFileName(string filePath)
             return filePath.EndsWith(DllExtension, StringComparison.OrdinalIgnoreCase);
         }
 
+        private void ApplyTaskOutput(TaskOutputs outputs)
+        {
+            RuleSetFilePath = outputs.Ruleset;
+            AnalyzerFilePaths = RemoveNonAnalyzerFiles(outputs.AssemblyPaths);
+            AdditionalFilePaths = outputs.AdditionalFilePaths;
+        }
+
         #endregion Private methods
+
+        /// <summary>
+        /// Internal data class to hold the set of output values for this task
```

Comment:
There are only three properties that need to be set so I could have either set them inline or used an anonymous type, but I decided this was cleaner and easier to read.
Also, collecting the values first then applying them at the end made it easier to perform common actions in a single place, such as removing non-assembly files.